src/vue-property-decorator.ts

```diff
@@ -26,13 +26,36 @@ export function Inject(key?: string | symbol): PropertyDecorator {
 }
 
 /**
+ * decorator of a provide
+ * @param key key
+ * @return PropertyDecorator | void
+ */
+export function Provide(key?: string | symbol): PropertyDecorator {
+  return createDecorator((componentOptions, k) => {
+		var provide: any = componentOptions.provide;
+		if('function'!== typeof provide || !provide.managed) {
```

Comment:
I couldn't understand how `componentOptions.provide.managed` works...
Instead of adding a new property, I recommend using `WeakMap` to manage something you'd like to do.

```ts
let container: WeakMap<ComponentOptions<Vue>, object> | null = null

export function Provide(key?: string | symbol): PropertyDecorator {
  // for the browsers doesn't support WeakMap
  // Both WeakMap and Symbol are ES2015 features.
  if (container == null) container = new WeakMap()
  return createDecorator((componentOptions, k) => {
    const managed = container.get(componentOptions)
    // ...
  }
}
```